src/theme/css/general.css

```diff
@@ -202,6 +202,10 @@ kbd {
     display: inline;
 }
 
+.footnote-reference{
+    line-height: 0;
```

Comment:
As far as I understand the standard, this does what you want because [a line box's final height is the maximum of the required line-height of all the inline fragment boxes within it](https://www.w3.org/TR/css-inline-3/#line-boxes). This means, if you set the line-height of something to 0, that element will have no effect on the height of the line.

It's fine, as long as the footnote-reference is never alone on a line. That would be bad typography anyway.

GitHub and rustdoc's CSS both do the exact same thing. You might want to apply this style to *all* superscripts, not just footnote references.

https://github.com/necolas/normalize.css/blob/master/normalize.css#L128